lib/motions/general-motions.coffee

```diff
@@ -289,6 +289,36 @@ class MoveToEndOfWord extends Motion
 class MoveToEndOfWholeWord extends MoveToEndOfWord
   wordRegex: WholeWordRegex
 
+class MoveToNextSentence extends Motion
+  moveCursor: (cursor, count=1) ->
+    _.times count, ->
+      start = cursor.getBufferPosition().translate new Point(0, 1)
+      eof = cursor.editor.getBuffer().getEndPosition()
+      scanRange = [start, eof]
+
+      cursor.editor.scanInBufferRange /(^$)|(([\.!?] )|^[A-Za-z0-9])/, scanRange, ({matchText, range, stop}) ->
```

Comment:
Also, after some testing, I've discovered that it is necessary to search for some character (maybe just a printable non-whitespace character) in order to accomplish the "next sentence on next line" edge case. I'm still looking to see if we don't need the A-Za-z or similar character sequence, but no luck so far.